app/views/rules/index.html.erb

```diff
@@ -32,19 +29,45 @@
     </p>
   </div>
 <% end %>
-
 <div class="bg-container shadow-border-xs rounded-xl p-4">
-
   <% if @rules.any? %>
-    <div class="rounded-xl bg-gray-25 space-y-1">
-      <div class="flex items-center gap-1.5 px-4 py-2 text-xs font-medium text-secondary uppercase">
-        <p>Rules</p>
-        <span class="text-subdued">&middot;</span>
-        <p><%= @rules.count %></p>
+    <div class="bg-container-inset rounded-xl">
+      <div class="flex justify-between px-4 py-2 text-xs uppercase">
+        <div class="flex items-center gap-1.5 font-medium text-secondary">
+          <p>Rules</p>
+          <span class="text-subdued">&middot;</span>
+          <p><%= @rules.count %></p>
+        </div>
+        <div class="flex items-center gap-1">
+          <span class="text-secondary pr-2">Sort by:</span>
+          <%= form_with url: rules_path, method: :get, local: true, class: "flex items-center gap-2", data: { controller: "auto-submit-form" } do |form| %>
+            <%= form.select :sort_by,
+                      options_for_select([["Name", "name"], ["Updated At", "updated_at"]], @sort_by),
+                      {},
+                      class: "min-w-[120px] bg-container rounded border-primary cursor-pointer text-primary text-sm w-auto",
+                      data: { auto_submit_form_target: "auto", autosubmit_trigger_event: "change" } %>
+            <%= form.hidden_field :direction, value: (@direction == "asc" ? "desc" : "asc") %>
+            <%= render ButtonComponent.new(
+                variant: :icon,
+                icon: "arrow-up-down",
+                size: :sm,
+                type: "submit",
+                title: "Toggle sort direction"
+              ) do %>
+              <span class="sr-only">Toggle sort direction</span>
+            <% end %>
+          <% end %>
```

Comment:
I think the easiest way to handle this would be to keep the `sort_by` select field inside the form and move the direction toggle outside the form as a link.

You could still keep the `form.hidden_field :direction` inside the form, and then you could additionally include the `sort_by` in the link:

```erb
<%= render LinkComponent.new(
   href: rules_path(direction: @direction == "asc" ? "desc" : "asc", sort_by: @sort_by)
) %>
```

That way, you let the URL deal with syncing the state between the two

Reply:
Awesome - fixed this!